lib/ui/progress.coffee

```diff
@@ -0,0 +1,30 @@
+{formatTimePeriod} = require '../misc'
+
+module.exports =
+  progs: []
+
+  add: (p) ->
+    pr = @ink.progress.create p
+    pr.register()
+    pr.t0 = Date.now()
+    @progs.push pr
+
+  update: (p) ->
+    pr = @progs.find (pr) => pr.id is p.id
```

Comment:
You mean make `@progs` an object (`{}`)  and then do `@progs[p.id]` instead of the `find` stuff?